app.go

```diff
@@ -186,11 +186,55 @@ func (a *App) useShortOptionHandling() bool {
 	return a.UseShortOptionHandling
 }
 
+func checkDuplicateSubcommandNames(subcommands []Command, parentCommandName string) error {
+	subcommandNamesWithoutDuplicates := make(map[string]bool)
+	for _, s := range subcommands {
+		_, sExists := subcommandNamesWithoutDuplicates[s.Name]
+		if sExists {
+			// @todo: update the following error message to take into account that the parent command may be a subcommand
+			return fmt.Errorf("Your command %q contains multiple subcommands with the Name %q. Having multiple subcommands with the same name results in ambiguous behavior, so please make sure each subcommand in your command has a unique name.", parentCommandName, s.Name)
```

Comment:
⭐⭐ _(seriously consider)_

```suggestion
			return fmt.Errorf("Your command %q contains multiple adjacent subcommands with the Name %q. Having multiple adjacent subcommands with the same name results in ambiguous behavior, so please make sure each adjacent subcommand within a command has a unique name.", parentCommandName, s.Name)
```

With respect to your `@todo`, subcommands are themselves commands so I don't think the error message needs to explicitly state something to the nature of _"within a command **or subcommand**"_. I could be convinced otherwise though 🙂